### PR TITLE
[Release Test] Send release test result to db pipeline for new test infra

### PR DESCRIPTION
**Pull request by jjyao**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ray_release/reporter/db.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            "buildkite_url": result.buildkite_url or "",
+            "runtime": result.runtime or -1.0,
+            "stable": result.stable,
+            "return_code": result.return_code,
+        }
+
+        logger.debug(f"Result json: {json.dumps(result_json)}")
+
+        try:
+            self.firehose.put_record(
```
Comment (jjyao) [inline release/ray_release/reporter/db.py:40]: I'm relying on boto3's built-in retry.

**Comment by jjyao** [comment]
It works: https://buildkite.com/ray-project/release-tests-branch/builds/69#_

File: release/ray_release/reporter/db.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                DeliveryStreamName="ray-ci-results",
+                Record={"Data": json.dumps(result_json)},
+            )
+        except Exception:
+            logger.exception("Failed to persist result to the databricks delta lake")
+
+        logger.info("Result has been persisted to the databricks delta lake")
```
Comment (krfricke) [inline release/ray_release/reporter/db.py:47]:
```suggestion
        else:
            logger.info("Result has been persisted to the databricks delta lake")
```

Comment (krfricke) [reply release/ray_release/reporter/db.py:47]: nit

**Review by krfricke — APPROVED** [review]
This is great!

**Comment by jjyao** [comment]
It's ready to be merged.

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**